app/uk/gov/hmrc/brm/utils/BrmException.scala

```diff
@@ -67,8 +71,15 @@ trait BRMException extends Controller {
       GatewayTimeout
   }
 
-  // TODO implement connection down for GRO
   def upstreamErrorPF(method: String)(implicit payload: Payload) : PartialFunction[Throwable, Result] = {
+    case e: Upstream5xxResponse if e.message.contains("GRO_CONNECTION_DOWN") =>
+      logException(method, s"ServiceUnavailable: ${e.getMessage}", SERVICE_UNAVAILABLE)
+      logException(method, s"Response body: ${ErrorResponse.GRO_CONNECTION_DOWN}", SERVICE_UNAVAILABLE)
+      ServiceUnavailable(ErrorResponse.GRO_CONNECTION_DOWN)
+    case e: Upstream5xxResponse if e.message.contains("SERVICE_UNAVAILABLE") =>
+      logException(method, s"ServiceUnavailable: ${e.getMessage}", SERVICE_UNAVAILABLE)
```

Comment:
@manishsw could we put these into their own partial functions and apply them in the filter. Makes it more readable.